### PR TITLE
Go version upgrades, base image upgrades & migration to GH actions

### PR DESCRIPTION
**Pull request by alexandreLamarre**
## Related issue 

https://github.com/rancherlabs/rancher-security/issues/1295

## Checklist

- [x] Update go versions
- [X] Vet packages for CVEs and upgrade as necessary (with upstream changes)
- [X] Move CI from drone to github actions
  - [x] Build & test
  - [X] Publish
  - [X] Remove build / CI dependencies on dapper (previously required to make cross-build image building easier)

### CODE REVIEW

File: .github/workflows/publish.yaml
Comment (alexandreLamarre) [inline .github/workflows/publish.yaml:1]: My working knowledge of github actions is not super in depth, so specifically for the release CI if we can make simplifications here while preserving the functionality of the original drone release pipeline that would be great

File: package/client/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM alpine:3.15.4
+FROM alpine:3.18
```
Comment (joshmeranda) [inline package/client/Dockerfile:1]: Would prefer pinning to a patch version if possible

File: scripts/build [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 set -e
-
+set -x
```
Comment (joshmeranda) [inline scripts/build:3]: Don't have much experience with our drone stuff, do we usually as `set -x`? If its not the standard seems like its gonna clutter the output a lot

**Review by joshmeranda — CHANGES_REQUESTED** [review]

File: .drone.yml.bak
[the remark this answers is not in the document]
Comment (alexandreLamarre) [reply .drone.yml.bak:1]: Intentionally keeping it as reference, at least temporarily

**Review by alexandreLamarre — COMMENTED** [review]
Reminder to update the docs & rancher changelogs before merging

File: .golangci.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
 		"disable-all": true,
 		"enable": [
 			"govet",
-			"golint",
+			"revive",
```
Comment (joshmeranda) [inline .golangci.json:6]: Is there a reason for using revive over goling?

**Review by joshmeranda — APPROVED** [review]

File: .golangci.json (hunk 3 above)
Comment (alexandreLamarre) [reply .golangci.json:6]: no longer officially supported, revive is the spiritual successor to that, at least according to golangcilint

**Pull request merged**